subprojects/packagefiles/glfw/meson.build

```
@@ -0,0 +1,221 @@
+endforeach
+cfg_file = configure_file(configuration : cfg_data,
+                          output : 'glfw_config.h')
+
+
+### Targets ###
+#@ Install key
+I = get_option('install') 
+#@ Primary target:
+##   GLFW library
+#  Common files
+s_common = [cfg_file,
+           'src/context.c',
+           'src/init.c',
+           'src/input.c',
+           'src/monitor.c',
+           'src/vulkan.c',
+           'src/window.c']
+#  Backend files
+if opt_display == 'osmesa'
+  s_display = ['src/null_init.c',
+               'src/null_monitor.c',
+               'src/null_window.c',
+               'src/null_joystick.c',
+               'src/posix_time.c',
+               'src/posix_thread.c',
+               'src/osmesa_context.c']
+elif opt_display == 'x11'
+  s_display = ['src/x11_init.c',
+               'src/x11_monitor.c',
+               'src/x11_window.c',
+               'src/xkb_unicode.c',
+               'src/posix_time.c',
+               'src/posix_thread.c',
+               'src/osmesa_context.c',
+               'src/egl_context.c',
+               'src/glx_context.c']
+  if sys_os == 'linux'
+    s_display += 'src/linux_joystick.c'
+  else
+    s_display += 'src/null_joystick.c'
+  endif
+elif opt_display == 'wayland'
+  s_display = [s_wayland,
+              'src/wl_init.c',
+              'src/wl_monitor.c',
+              'src/wl_window.c',
+              'src/xkb_unicode.c',
+              'src/posix_time.c',
+              'src/posix_thread.c',
+              'src/osmesa_context.c',
+              'src/egl_context.c']
+  if sys_os == 'linux'
+    s_display += 'src/linux_joystick.c'
+  else
+    s_display += 'src/null_joystick.c'
+  endif
+elif opt_display == 'win32'
+#——TODO:
+# ☐ Implement Win32 build routine
+  s_display = ['src/win32_init.c',
+               'src/win32_joystick.c',
+               'src/win32_monitor.c',
+               'src/win32_time.c',
+               'src/win32_thread.c',
+               'src/win32_window.c',
+               'src/wgl_context.c']
+elif opt_display == 'cocoa'
+#——TODO:
+# ☐ Implement Cocoa build routine
+  s_display = ['src/cocoa_init.m',
+               'src/cocoa_joystick.m',
+               'src/cocoa_monitor.m',
+               'src/cocoa_window.m',
+               'src/cocoa_time.c',
+               'src/posix_thread.c',
+               'src/nsgl_context.m']
+endif
+srcfiles = [s_common, s_display]
+incdirs = include_directories('include', 'src')
+#  Build
+glfw_lib = library('glfw3',
+                    srcfiles,
+                    c_args : ['-D_GLFW_USE_CONFIG_H'],
+                    include_directories : incdirs,
+                    dependencies : deps,
+                    version : meson.project_version(),
+                    build_by_default : true,
+                    pic : true,
+                    install : I)
+glfw_dep = declare_dependency(include_directories : incdirs, link_with : glfw_lib)
+meson.override_dependency('glfw3', glfw_dep)
```

If this was wrapped in

```suggestion
if meson.version().version_compare('>=0.54.0')
    meson.override_dependency('glfw3', glfw_dep)
endif
```

then the minimum version of Meson could drop to 0.49.0 instead.